MarkdownToLatex/MarkdownToLatex/LatexRenderer.cs

```diff
@@ -1,7 +1,204 @@
+            }
+        }}
+
+        private static byte _inquote;
+
+        /// <summary>a byte indicating whether a quote is currently rendered.</summary>
+        /// <exception cref="System.FormatException">Thrown when trying to set the value to something other than 0-3.</exception>
+        public static byte InQuote {get => _inquote; private set {
+            if(0 <= value && value <= 3){
+                _inquote = value;
+            } else {
+                throw new FormatException("Input has to be between 0 and 3. Check the documentation for details."); 
+            }
+        }}
+
+        /// <summary>Writes the LaTeX document into a file at the specified <paramref name="path"/>.</summary>
+        public static void WriteLatexDocument(string path){
+            File.WriteAllLines(path, LatexLines);
+        }
+
+        /// <summary>Writes a MathElement in LaTeX.</summary>
+        public static void WriteMathElement(string line){
+            LatexLines.Add(line); //The calculation stuff is handled in MdToTex class
+        }
+
+        /// <summary>Writes a Headline in LaTeX using a Match <paramref name="m"/>.</summary>
+        public static void WriteHeadline(Match m){
+            if(m == Match.Empty) {return;} //SHOULD never happen, but you never know ;D
+
+            int length = m.Groups[1].Value.Length; //First Group, number of #'s
+            string caption = m.Groups[2].Value; //Second Group, the actual text
+
+            switch(length){
+                case 0:
+                    LatexLines.Add(caption); //Should also not happen
+                    break;
+                case 1:
+                    LatexLines.Add(String.Format(@"\chapter*{{{0}}}", caption));
+                    break;
+                case 2:
+                    LatexLines.Add(String.Format(@"\section*{{{0}}}", caption));
+                    break;
+                case 3:
+                    LatexLines.Add(String.Format(@"\subsection*{{{0}}}", caption));
+                    break;
+                default:
+                    LatexLines.Add(String.Format(@"\subsection*{{{0}}}", caption)); //Treat all further headlines as subsections in LaTeX
+                    break;
+            }
+        }
+
+        /// <summary>Writes a List in LaTeX using a Match <paramref name="m"/></summary>
+        public static void WriteList(Match m){
+            int depth = m.Groups[1].Value.Length/2+1;
+            string content = m.Groups[3].Value;
+
+            if(m == Match.Empty) depth = 0;
+
+            if(depth < InList){
+                InList = 0;
+            }
+
+            if(InList == 0 && depth == 1){ //if not in any list
+                LatexLines.Add(@"\begin{itemize}");
+                LatexLines.Add(String.Format(@"\item{{{0}}}", content));
+                LatexLines.Add(@"\end{itemize}");
+                InList = 1;
+            } else if (InList == 1){
```

Comment:
Im Zusammenhang mit der `while` Schleife von oben:  
Es braucht eine Möglichkeit nur eine Ebene runter zu steigen. Wir müssen also wissen was die **alte** `InList` war.  

Vorschlag mit Bsp.:  
- Wir sind in Ebene 2 und bekommen eine Zeile mit Ebene 1
- `while` Schleife sorgt dafür, dass depth = InList = 1
- ***neues privates Feld*** `oldInList`, welches am besten immer nach dem if-Baum den aktuellen `InList`-Wert annimmt = 2
- `newListAfterItem()` --> `newList()` + zusätzllichen `int` Parameter (einen Offset [0 - after item, 1 - index+1, 2 - index+2, ...])

``` C#
if (depth == 1 && oldInList > InList) {
    newList(content, oldList-InList);
} else if (depth == 1 && oldInList == InList) {
    appendNewItem(content);
} else {
    newList(content, 0);
}
```

- oldInList > InList --> `Insert` neue Liste beim Index des letzten Items + 2